p2p/listener.go

```diff
@@ -176,26 +174,29 @@ func (l *ListenHandler) handleStream(stream network.Stream) {
 			randPeerID, randListeners, randTs, err := announce.GetRandom(reqID)
 			var randData [][]byte
 			if nil != err || util.IDEqual(reqID, randPeerID) { // No Random Node sendback this Node
-				randData, err = PackRegisterData(nodeChain, fn, nType, reqID, reqMaAddrs, time.Now())
+				randData, _ = PackRegisterData(nodeChain, fn, nType, reqID, reqMaAddrs, time.Now())
 				util.LogDebug(log, util.CoReset, fmt.Sprintf("Send back peer as a random node ID:%v addrs:%v", reqID.ShortString(), util.PrintMaAddrs(reqMaAddrs)))
 			} else { //Get a Random Node
-				randData, err = PackRegisterData(nodeChain, fn, nType, randPeerID, randListeners, randTs)
+				randData, _ = PackRegisterData(nodeChain, fn, nType, randPeerID, randListeners, randTs)
 				util.LogDebug(log, util.CoReset, fmt.Sprintf("Send a random node ID:%v addrs:%v", randPeerID.ShortString(), util.PrintMaAddrs(randListeners)))
 			}
-
+			if 0 == len(randData) { //deal with pack error here not above
+				listenerSendError(rw, nodeChain, fault.PackRandomNodeFail, "-><- Radom node", log)
```

Comment:
`listenerSendError` sometimes `break` and sometimes not. Do you intend to do this for further logging?

Reply:
all should be break